app/src/main/java/hr/foi/air/servicesync/ui/contents/CompanyDetailsContent.kt

```diff
@@ -27,54 +23,60 @@ import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
 import com.example.compose.onSurfaceDark
 import com.example.compose.onSurfaceLight
 import com.google.firebase.firestore.GeoPoint
 import hr.foi.air.servicesync.R
 import hr.foi.air.servicesync.backend.FirestoreCompanyDetails
 import hr.foi.air.servicesync.ui.components.CompanyDescription
+import hr.foi.air.servicesync.ui.components.CompanyImage
 import hr.foi.air.servicesync.ui.components.CompanyLocation
-import hr.foi.air.servicesync.ui.components.CompanyNameAndImage
 import hr.foi.air.servicesync.ui.components.isDark
 import hr.foi.air.servicesync.ui.items.ProvidedServicesListItem
 import mapproviders.GoogleMapProvider
-import mapproviders.OpenStreetMapProvider
 
 @Composable
 fun CompanyDetailsContent(
     modifier: Modifier = Modifier,
-    context: Context
+    context: Context,
+    navController: NavController,
+    companyName: String
 ) {
     val firestoreCompanyDetails = FirestoreCompanyDetails()
 
-    val companyName = remember { mutableStateOf("Loading...") }
     val companyDescription = remember { mutableStateOf("Loading...") }
     val companyCategory = remember { mutableStateOf("Loading...") }
     val companyWorkingHours = remember { mutableStateOf(0) }
-    val companyGeoPoint = remember { mutableStateOf(GeoPoint(0.0, 0.0)) }
+    val companyGeoPoint = remember { mutableStateOf<GeoPoint?>(null) }
+    val companyImageUrl = remember { mutableStateOf<String?>(null) }
     val isLoading = remember { mutableStateOf(true) }
 
-    LaunchedEffect(Unit) {
-        firestoreCompanyDetails.loadCompanyName(context) { name ->
-            companyName.value = name ?: "No name found!"
-            isLoading.value = false
-        }
-        firestoreCompanyDetails.loadCompanyDescription(context) { description ->
+    LaunchedEffect(companyName) {
```

Comment:
These things should be in their separate layer. Please consider this in the next/further sprint/s